onnx/defs/nn/defs.cc

```diff
@@ -2916,24 +2916,24 @@ ONNX_OPERATOR_SET_SCHEMA(
             "interleaved",
             "Rotate using interleaved pattern. Default value is 0 (False).",
             AttributeProto::INT,
-            OPTIONAL_VALUE)
+            static_cast<int64_t>(0))
         .Attr(
             "rotary_embedding_dim",
             "Rotary embedding dimension used to apply partial rotary embeddings.",
             AttributeProto::INT,
-            OPTIONAL_VALUE)
+            static_cast<int64_t>(0))
         .Attr(
             "num_heads",
-            "Number of attention heads. Must use with `rotary_embedding_dim`. ",
+            "Number of attention heads. Must be provided when input is a 3D tensor. ",
             AttributeProto::INT,
-            OPTIONAL_VALUE)
+            static_cast<int64_t>(0))
```

Comment:
Wait ... this is an optional attribute, right? A value of 0 doesn't mean anything for num_heads. Better to leave it optional, with no default value.

Reply:
Or, a default value of 1 might be ok, for the single-head case